manager/integration/tests/test_recurring_job.py

```diff
@@ -636,4 +683,64 @@ def test_recurring_jobs_when_volume_detached_unexpectedly():  # NOQA
     10. Turn off `Allow Recurring Job While Volume Is Detached` setting
        Clean up backups, volumes
     """
-    pass
+
+    recurring_job_setting = \
+        client.by_id_setting(SETTING_RECURRING_JOB_WHILE_VOLUME_DETACHED)
+    client.update(recurring_job_setting, value="true")
+
+    pvc_name = 'pvc-volume-detached-unexpectedly-test'
+    pvc['metadata']['name'] = pvc_name
+    pvc['spec']['storageClassName'] = 'longhorn'
+
+    core_api.create_namespaced_persistent_volume_claim(
+        body=pvc, namespace='default')
+
+    deployment = make_deployment_with_pvc(
+        'deployment-volume-detached-unexpectedly-test', pvc_name)
+    create_and_wait_deployment(apps_api, deployment)
+    pod_names = common.get_deployment_pod_names(core_api, deployment)
+    vol_name = get_volume_name(core_api, pvc_name)
+
+    write_pod_volume_random_data(core_api, pod_names[0], "/data/test",
+                                 DATA_SIZE_IN_MB_2)
+
+    deployment['spec']['replicas'] = 0
+    apps_api.patch_namespaced_deployment(body=deployment,
+                                         namespace='default',
+                                         name=deployment["metadata"]["name"])
+    vol = wait_for_volume_detached(client, vol_name)
+
+    jobs = [
+        {
+            "name": RECURRING_JOB_NAME,
+            "cron": "*/2 * * * *",
+            "task": "backup",
+            "retain": 1
+        }
+    ]
+    vol.recurringUpdate(jobs=jobs)
+
+    wait_for_recurring_backup_to_start(client, core_api, vol_name, 1)
+
+    crash_engine_process_with_sigkill(client, core_api, vol_name)
+    time.sleep(30)
+    wait_for_volume_attached(client, vol_name)
```

Comment:
Is it better to wait for the volume to be in the attached state or to be healthy as in the test skeleton?

Reply:
fixed